CONTRIBUTING.md

```diff
@@ -1,3 +1,103 @@
+Building and debugging process is described [here](https://wiki.jenkins-ci.org/display/JENKINS/Building+Jenkins).
+
+If you want simply to have the `jenkins.war` file as fast as possible without tests, run:
+
+    mvn clean package -pl war -am -DskipTests
+
+The WAR file will be created in `war/target/jenkins.war`. 
+After that you can just start Jenkins using Java CLI ([guide](https://wiki.jenkins.io/display/JENKINS/Starting+and+Accessing+Jenkins)).
+If you want to debug this WAR file without using Maven plugins,
+You can just start the executable with [Remote Debug Flags](https://stackoverflow.com/questions/975271/remote-debugging-a-java-application) 
+and then attach IDE Debugger to it.
+
+## Testing changes
+
+Jenkins core offers unit and functional tests as a part of the repository.
+
+Functional tests (`test` module) take a while even on server-grade machines.
+The most of the tests will be launched by the continuous integration instance,
+so there is no strict need to run full test suites before proposing a pull request.
+
+In addition to the included tests, you can also find extra integration and UI 
+tests in the [Acceptance Test Harness (ATH)](https://github.com/jenkinsci/acceptance-test-harness) repository.
+If you propose complex UI changes, it is advised to create new ATH tests for them.
+
+## Proposing Changes
+
+Jenkins project uses GitHub as a main engine for patch proposals and code reviews.
+All proposals should be submitted as pull requests.
+In order to do that...
+
+1. Commit changes and push them to your fork on GitHub. 
+It is a good practice is to create branches instead of pushing to master.
+2. In GitHub Web UI click the _New Pull Request_ button
+3. Select `jenkinsci` as _base fork_ and `master` as `base`, then click _Create Pull Request_
+  * We integrate all changes into the master branch towards the Weekly releases
+  * After that the changes may be backported to the current LTS baseline by the LTS Team.
+    The backporting process is described [here](https://jenkins.io/download/lts/).
+4. Fill in the Pull Request description according to the [proposed template](.github/PULL_REQUEST_TEMPLATE.md).
+5. Click _Create Pull Request_
+6. Wait for CI results/reviews, process the feedback.
+  * If you do not get feedback after 3 days, feel free to ping `@jenkinsci/code-reviewers` to CC.
+  * Usually we merge pull requests after 2 votes from reviewers or after 1 vote and 1-week delay without extra reviews.
+
+Once your Pull Request is ready to be merged, 
+the repository maintainers will integrate it, prepare changelogs and 
+ensure it gets released in one of incoming Weekly releases.
+There is no extra action items required from pull request authors at this point.
+
+## Copyright
+
+Jenkins core is licensed under MIT license, with few exceptions in the bundled classes.
+
+We **Do NOT** require pull request submitters to sign the [contributor agreement](https://wiki.jenkins.io/display/JENKINS/Copyright+on+source+code)
```

Comment:
Perhaps make it explicit that all contributions are MIT unless otherwise stated? IANAL though, not sure how that is supposed to work.

Reply:
I would push-back non-MIT code proposals unless it's ultimately necessary.
Yes, it makes sense to make it explicit